parastell/utils.py

```diff
@@ -101,3 +123,73 @@ def filter_kwargs(
         raise e
 
     return {name: dict[name] for name in allowed_keys}
+
+
+def normalize(vec_list):
+    """Normalizes a set of vectors.
+
+    Arguments:
+        vec_list (1 or 2D np array): single 1D vector or array of 1D vectors
+            to be normalized
+    Returns:
+        vec_list (np array of same shape as input): single 1D normalized vector
+            or array of normalized 1D vectors
+    """
+    if len(vec_list.shape) == 1:
+        return vec_list / np.linalg.norm(vec_list)
+    elif len(vec_list.shape) == 2:
+        return vec_list / np.linalg.norm(vec_list, axis=1)[:, np.newaxis]
+    else:
+        print('Input "vec_list" must be 1-D or 2-D NumPy array')
+
+
+def read_yaml_config(filename):
+    """Read YAML file describing ParaStell configuration and extract all data."""
+    with open(filename) as yaml_file:
+        all_data = yaml.safe_load(yaml_file)
+
+    return all_data
+
+
+def reorder_loop(list, index):
+    """Reorders a list representing a closed loop.
+
+    Arguments:
+        list (iterable): closed loop.
+        index (int): list index about which to reorder loop.
+
+    Returns:
+        reordered_loop (iterable): reordered closed loop.
+    """
+    reordered_list = np.concatenate([list[index:], list[1:index]])
+    reordered_list = np.append(reordered_list, [reordered_list[0]], axis=0)
```

Comment:
Is there a case (edge case?) where this doesn't work?
```suggestion
    reordered_list = np.concatenate([list[index:], list[1:index+1]])
```
maybe if  `index == len(list)`?

Reply:
Yes, in that case this would result in an index error

Reply:
Actually, this would work even in that edge case since `index+1` would never actually be reached.